diff --git a/node_modules/react-native-multi-ble-peripheral/android/src/main/java/com/fugood/reactnativemultibleperipheral/ReactNativeMultiBlePeripheralModule.kt b/node_modules/react-native-multi-ble-peripheral/android/src/main/java/com/fugood/reactnativemultibleperipheral/ReactNativeMultiBlePeripheralModule.kt
index a05fd62..884d9ef 100644
--- a/node_modules/react-native-multi-ble-peripheral/android/src/main/java/com/fugood/reactnativemultibleperipheral/ReactNativeMultiBlePeripheralModule.kt
+++ b/node_modules/react-native-multi-ble-peripheral/android/src/main/java/com/fugood/reactnativemultibleperipheral/ReactNativeMultiBlePeripheralModule.kt
@@ -238,11 +238,12 @@ class ReactNativeMultiBlePeripheralModule(reactContext: ReactApplicationContext)
     var value = Base64.decode(value, Base64.DEFAULT)
     characteristic.value = value
     for (device in registeredDevices) {
-      val response = bluetoothGattServer.notifyCharacteristicChanged(
-        device,
-        characteristic,
-        confirm
-      )
+      val response = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        bluetoothGattServer.notifyCharacteristicChanged(device, characteristic, confirm, value)
+      } else {
+        @Suppress("DEPRECATION")
+        bluetoothGattServer.notifyCharacteristicChanged(device, characteristic, confirm)
+      }
     }
     promise.resolve(null)
   }
@@ -322,10 +323,10 @@ class ReactNativeMultiBlePeripheralModule(reactContext: ReactApplicationContext)
           var gattServer = bluetoothGattServers[id]
           if (gattServer != null) {
             val params = Arguments.createMap()
-            params.putString("id", id.toString())
+            params.putInt("id", id)
             params.putString("device", device.address)
-            params.putString("service", characteristic.service.uuid.toString())
-            params.putString("characteristic", characteristic.uuid.toString())
+            params.putString("serviceUuid", characteristic.service.uuid.toString())
+            params.putString("characteristicUuid", characteristic.uuid.toString())
             params.putInt("offset", offset)
             params.putString("value", Base64.encodeToString(value, Base64.NO_WRAP))
             sendEvent("onWrite", params)
diff --git a/node_modules/react-native-multi-ble-peripheral/ios/ReactNativeMultiBlePeripheral.swift b/node_modules/react-native-multi-ble-peripheral/ios/ReactNativeMultiBlePeripheral.swift
index 46cdbb0..39d0e92 100644
--- a/node_modules/react-native-multi-ble-peripheral/ios/ReactNativeMultiBlePeripheral.swift
+++ b/node_modules/react-native-multi-ble-peripheral/ios/ReactNativeMultiBlePeripheral.swift
@@ -305,5 +305,6 @@ class ReactNativeMultiBlePeripheral: RCTEventEmitter, CBPeripheralManagerDelegat
         "characteristicUUID": characteristicUUID,
         "value": base64,
       ])
+      peripheral.respond(to: request, withResult: .success)
     }
   }
